server/controllers/aws/Logs/getLogs.js

```diff
@@ -74,54 +78,70 @@ const getLogs = async (req, res, next) => {
-        attempts: 1,
-        totalRetryDelay: 0 
-      },
-      events: [],
-      nextToken: undefined,
-      searchedLogStreams: [] 
-    }
-
-  */
-		// if no log events, just go back to frontend
+
+		// if no log events exist, return back to frontend
 		if (!logEvents) {
 			res.locals.functionLogs = false;
 			return next();
 		}
-		// only send back most recent 50 logs to reduce size
+		// only send back most recent 50 logs to reduce size of payload
 		const shortenedEvents = [];
 
-		// if we received a nextToken, start helperFunc process and make sure to parse through that data in order to grab from the end
+		// if we received a nextToken, start helperFunc to recursively parse through most recent data (meaning we grab data from the end since that is the most recent log stream)
 		if (logEvents.nextToken) {
 			const helperFuncResults = await helperFunc(logEvents.nextToken);
+
+			// poppedEl gets the most recent log stream that currently exists in helperFunc (log streams that are even more recent will have already been added to shortenedEvents)
 			let poppedEl;
 
-			// while we still have logs to grab from the helperFunc and shortenedEvents is shorter than 50 logs, add to it from the end (giving us the most recent first instead)
+			// while we still have logs to grab from the helperFunc and shortenedEvents is shorter than 50 logs, add to shortenedEvents array from the end (the most recent log stream)
 			while (
 				helperFuncResults.length &&
 				shortenedEvents.length <= 50
 			) {
+				// poppedEl gets the most recent log stream that currently exists in helperFunc (log streams that are even more recent will have already been added to shortenedEvents)
+				// but the for loop below is iterating through helperFunc such that we are adding the most recent log stream at the beginning of the shortenedEvent array
 				poppedEl = helperFuncResults.pop();
+				/**
+				
+				shortenedEvent = [                                              helperFuncResults = [
+					index 0: { most recent event log stream },                      index 0: { least recent event log stream }
+					.                                                               .
+					.                                                               .
+					.                                                               .
+					index N: { least recent event log stream },                     index N: { most recent event log stream }
+				]																																]
+				
+				*/
 				for (let i = poppedEl.length - 1; i >= 0; i -= 1) {
-					if (shortenedEvents.length === 50) {
-						break;
-					}
-					shortenedEvents.push(poppedEl[i]);
+					// we don't want to have more than 50 logs at any point in time to reduce operational load and size
+					if (shortenedEvents.length === 50) break;
+					else shortenedEvents.push(poppedEl[i]);
 				}
 			}
 		}
+		/**
+		 *
+		 * If we didn't have a nextToken and got all logs in one request to the CloudWatchLogsClient
```

Comment:
I would clean this up -- the conditional can warrant a couple sentences maybe, but not a couple paragraphs : )